discovery-provider/src/utils/redis_metrics.py

```diff
@@ -64,10 +66,11 @@ def extract_app_name_key():
     """
     application_name = request.args.get(app_name_param, type=str, default=None)
     ip = request.headers.get('X-Forwarded-For', request.remote_addr)
-    ip = ip.split(',')[0].strip()
+    # Replace the `:` character with an `_`  because we use : as the redis key delimiter
+    ip = ip.split(',')[0].strip().replace(":", "_")
```

Comment:
Can we make a util function that formats the IP? Don't the two use-cases to diverge in the future etc.

Reply:
+1